network.go

```diff
@@ -273,6 +277,129 @@ func getInterface(netHandle *netlink.Handle, link netlink.Link) *pb.Interface {
 // Routes //
 ////////////
 
+//updateRoutes will take requestedRoutes and create netlink routes, with a goal of creating a final
+// state which matches the requested routes.  In doing this, preesxisting non-loopback routes will be
+// removed from the network.  If an error occurs, this function returns the list of routes in
+// gRPC-route format at the time of failure
+func (s *sandbox) updateRoutes(netHandle *netlink.Handle, requestedRoutes *pb.Routes) (resultingRoutes *pb.Routes, err error) {
+
+	if netHandle == nil {
+		netHandle, err = netlink.NewHandle()
+		if err != nil {
+			return nil, err
+		}
+		defer netHandle.Delete()
+	}
+
+	//
+	// First things first, let's blow away all the existing routes.  The updateRoutes function
+	// is designed to be declarative, so we will attempt to create state matching what is
+	// requested, and in the event that we fail to do so, will return the error and final state.
+	//
+
+	initRouteList, err := netHandle.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return nil, err
+	}
+	for _, initRoute := range initRouteList {
+		// don't delete routes associated with lo:
+		link, _ := netHandle.LinkByIndex(initRoute.LinkIndex)
+		if link.Attrs().Name == "lo" || link.Attrs().Name == "::1" {
+			continue
+		}
+
+		err = netHandle.RouteDel(&initRoute)
+		if err != nil {
+			//If there was an error deleting some of the initial routes,
+			//return the error and the current state of the routes on the
+			//system
+			goto error_case
+		}
+	}
+
+	//
+	// Set each of the requested routes
+	//
+	// First make sure we set the interfaces initial routes, as otherwise we
+	// won't be able to access the gateway
+	for _, reqRoute := range requestedRoutes.Routes {
+		if reqRoute.Gateway == "" {
+			err = s.updateRoute(netHandle, reqRoute, true)
+			if err != nil {
+				agentLog.WithError(err).Error("update Route failed")
+				//If there was an error setting the route, return the
+				//error and the current list of routes on the system
+				goto error_case
+			}
+
+		}
+	}
+	// Take a second pass and apply the routes which include a gateway
+	for _, reqRoute := range requestedRoutes.Routes {
+		if reqRoute.Gateway != "" {
+			err = s.updateRoute(netHandle, reqRoute, true)
+			if err != nil {
+				agentLog.WithError(err).Error("update Route failed")
+				//If there was an error setting the route, return the
+				//error and the current list of routes on the system
+				goto error_case
+			}
+		}
+	}
+
+error_case:
```

Comment:
No need for `goto` here, and this code looks like C now lol.
I think doing something like this would work:
```go
func (s *sandbox) updateRoutes(netHandle *netlink.Handle, requestedRoutes *pb.Routes) (resultingRoutes *pb.Routes, err error) {
        defer func() {
                if err != nil {
                        routes, _ = getCurrentRoutes(netHandle)
                }
        }()

        ...
```
You only need to replace your calls to `goto error_case` with `return`

Reply:
I was taught in school to never use gotos.  And here I am.  Let me checkout what I can do for defer.